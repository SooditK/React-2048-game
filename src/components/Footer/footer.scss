@import '../../assets/styles/vars';

.footer {
  border-top: 1px $border-color solid !important;
  color: $black;
  font-size: 1.25rem;
  margin-top: 3.08vw;
  width: 100%;

  .container {
    align-items: center;
    display: flex;
    justify-content: center;
    margin: 3.08vw 0 2vw;
  }

  .icon {
    margin-right: 10px;
  }

  .github {
    width: 32px;
  }

  .heart {
    color: $peach;
    font-size: 1rem;
    margin: 0 10px;
    transform: scale(0.9);
  }

  .link {
    margin-left: 10px;
    outline: none !important;
    text-decoration: none !important;

    &:hover,
    &:active,
    &:focus {
      outline: none !important;
    }
  }
}

@media all and (max-width: $break-point) {
  .footer {
    margin-top: 8vw;
    width: 85vw;
  }
}
