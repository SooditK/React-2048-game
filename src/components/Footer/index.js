import React from 'react';
import PropTypes from 'prop-types';
import styles from './footer.scss';
import github from '../../assets/images/github.png';

// Footer, link to github repo and developer profile
export default class Footer extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    repoUrl: PropTypes.string.isRequired,
    profileUrl: PropTypes.string.isRequired
  };

  // Render once, no update
  shouldComponentUpdate() {
    return false;
  }

  render() {
    const {
      props: { name, repoUrl, profileUrl }
    } = this;

    return (
      <div className={styles.footer}>
        <div className={styles.container}>
          <div>
            <a href={repoUrl} className={styles.icon}>
              <img src={github} className={styles.github} alt="github repo" />
            </a>
          </div>
          Build with
          <span className={styles.heart}>â™¥</span>
          by
          <a href={profileUrl} className={styles.link}>
            {name}
          </a>
        </div>
      </div>
    );
  }
}
