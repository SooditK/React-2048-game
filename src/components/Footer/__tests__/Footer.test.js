import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Footer from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Footer />', () => {
  it('component render', () => {
    const footer = renderer
      .create(
        <Footer
          name="test"
          repoUrl="http://test.com"
          profileUrl="http://profile.test"
        />
      )
      .toJSON();
    expect(footer).toMatchSnapshot();
  });

  it('shouldComponentUpdate', () => {
    const footer = shallow(
      <Footer
        name="test"
        repoUrl="http://test.com"
        profileUrl="http://profile.test"
      />
    ).instance();
    expect(footer.shouldComponentUpdate()).toBe(false);
  });
});
