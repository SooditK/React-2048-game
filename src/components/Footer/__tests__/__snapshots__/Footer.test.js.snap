// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component render 1`] = `
<div
  className="footer"
>
  <div
    className="container"
  >
    <div>
      <a
        className="icon"
        href="http://test.com"
      >
        <img
          alt="github repo"
          className="github"
          src="github.png"
        />
      </a>
    </div>
    Build with
    <span
      className="heart"
    >
      â™¥
    </span>
    by
    <a
      className="link"
      href="http://profile.test"
    >
      test
    </a>
  </div>
</div>
`;
